module3-crud-and-data-models/r1.1-principles-of-setting-up-your-schema/README.md

```diff
@@ -0,0 +1,50 @@
+## Principles of setting up your schema
+
+Poorly designed databases can cause many problems, including a waste of resources, difficult maintenance, and faulty performance. That's why having a great database schema design is a crucial part of effective data management.
+
+There are few things to focus on when creating a database schema:
+
+- How the data will be stored (what database will be used)?
+- The query performance.
+- How much hardware it will take?
+
+### Type of database
+
+Most developers don't see the difference between relation database schema and MongoDB schema, but in reality, it is not the same
+
+<img src="https://i.imgur.com/3yobiPB.jpg" alt="meme" width="400"/>
+
+In a nutshell,
+
+in `relational databases`, models are usually independent of queries, there is no duplication of data as data will _mostly_ be separated into different tables, and it is rigid, you will have to define types and fields for your schema beforehand.
+
+in the other hand, `MongoDB`, you have more flexibility, there are no rules, documents can have new data or fields at any point of time, no need to define types.
+
+### Embedding vs. Referencing
+
+One of the key points to establish a good schema design (especially, if you are using MongoDB) is weither to embed or reference your data, as it can make a big difference in the performance and hardware use.
```

Comment:
whether or either, not sure which one you intended

Reply:
does it actually make a big difference in hardware use?

i'd google "premature optimization"

it's highly unlikely a student should be thinking too much about performance (i actually discourage students from doing so due to premature optimization)